source/rpt/particle_detector_interactions.cc

```diff
@@ -0,0 +1,454 @@
+#include <rpt/particle_detector_interactions.h>
+#include <stdlib.h>
+
+#include <cmath>
+
+template <int dim>
+void
+ParticleDetectorInteractions<dim>::calculate_position_parameters()
+{
+  // Vector from the detector face center to the particle position (Xdp)
+  Tensor<1, dim> face_detector_particle_distance =
+    particle_position - detector_face_position;
+
+  // Detector orientation vector parallel to z' in particle-detector coordinate
+  // (e'z)
+  Tensor<1, dim> face_middle_vector =
+    detector_face_position - detector_middle_position;
+  detector_orientation_z = face_middle_vector / face_middle_vector.norm();
+
+  // Projection of the distance vector on the detector axis which is parallel to
+  // the z' axis Particle position in particle-detector coordinate frame (h =
+  // z'p)
+  h = scalar_product(face_detector_particle_distance, detector_orientation_z);
+
+  // x' coordinate of the particle-detector frame (e'x) (see equations 29 to 31)
+  Tensor<1, dim> Z = h * detector_orientation_z;
+  Tensor<1, dim> X = Z - face_detector_particle_distance;
+
+  if (X.norm() == 0) // When particle is aligned to the detector
```

Comment:
Not a good idea to compare floats/double using equation. Maybe define a threshold?